website/www/site/content/en/documentation/io/built-in/snowflake.md

```diff
@@ -0,0 +1,738 @@
+  String getLoginTimeout();
+
+  void setLoginTimeout(String loginTimeout);
+
+  @Description("Temporary GCS bucket name.")
+  ValueProvider<String> getStagingBucketName();
+
+  void setStagingBucketName(ValueProvider<String> stagingBucketName);
+
+  @Description("Storage integration name")
+  ValueProvider<String> getStorageIntegrationName();
+
+  void setStorageIntegrationName(ValueProvider<String> storageIntegrationName);
+
+  @Description("SnowPipe name. Optional.")
+  ValueProvider<String> getSnowPipe();
+
+  void setSnowPipe(ValueProvider<String> snowPipe);
+}
+{{< /highlight >}}
+### Using Pipeline options
+To use Pipeline options, you must configure them as follows:
+
+{{< highlight java>}}
+SnowflakePipelineOptions options = PipelineOptionsFactory
+    .fromArgs(args)
+    .withValidation()
+    .as(SnowflakePipelineOptions.class);
+{{< /highlight >}}
+
+All the below parameters are required:
+- `.fromArgs()`
+  - GNU style command line arguments, 
+  - Example: `--project=myproject --database=test`
+- `.withValidation()`
+  - which validates that PipelineOptions confirms all criteria from the passed in interface
+- `.as()`
+  - a class of used pipeline options
+  - Example: `.as(SnowflakePipelineOptions.class)`
+
+Then create your pipeline using created options:
+
+{{< highlight java >}}
+Pipeline pipeline = Pipeline.create(options);
+{{< /highlight >}}
+
+Example of accessing pipeline options in code:
+
+{{< highlight java >}}
+String stagingBucketName = options.getStagingBucketName();
+{{< /highlight >}}
+
+Pipelines in tests:
+{{< highlight java >}}
+PipelineOptionsFactory.register(SnowflakePipelineOptions.class);
+    options = TestPipeline
+                .testingPipelineOptions()                                  
+                .as(SnowflakePipelineOptions.class);
+{{< /highlight >}}
+
+Then use them in pipeline run:
+{{< highlight java >}}
+PipelineResult pipelineResult = pipeline.run(options);
+{{< /highlight >}}
+
+### Extending pipeline options
+Extend the SnowflakePipelineOptions with your own custom options to access additional parameters in your code.
+
+Example of extending the Pipeline options:
+{{< highlight java >}}
+public interface BatchTestPipelineOptions extends SnowflakePipelineOptions {
+  @Description("Table name to connect to.")
+  String getTable();
+
+  void setTable(String table);
+}
+{{< /highlight >}}
+
+The following example shows how to insert the custom options:
+{{< highlight java >}}
+ExampleSnowflakePipelineOptions options = PipelineOptionsFactory
+.fromArgs(args)
+.withValidation()
+.as(ExampleSnowflakePipelineOptions.class);
+
+options.getTable();
+{{< /highlight >}}
+
+And in case of testing:
+{{< highlight java >}}
+PipelineOptionsFactory.register(ExampleSnowflakePipelineOptions.class);
+    options = TestPipeline
+    .testingPipelineOptions()
+                .as(ExampleSnowflakePipelineOptions.class);
+{{< /highlight >}}
+
+### Running main command with Pipeline options
```

Comment:
I am not sure if it should stay here? @DariuszAniszewski @purbanow

Reply:
To be honest I'm not sure. I think we can leave it for now.